quint/test/effects/inferrer.test.ts

```diff
@@ -158,22 +148,39 @@ describe('inferEffects', () => {
       'def a(tup) = Set(tup, (x, 4)).map((p, g) => p + g)',
     ])
 
-    const inferrer = new EffectInferrer(definitionsTable)
-    const [errors, effects] = inferrer.inferEffects(quintModule)
+    const [errors, effects] = inferEffectsForModule(quintModule)
 
     const expectedEffect = "∀ v0, v1 . (Read[v0] & Temporal[v1]) => Read[v0, 'x'] & Temporal[v1]"
 
     assert.isEmpty(errors, `Should find no errors, found: ${[...errors.values()].map(errorTreeToString)}`)
     assert.deepEqual(effectSchemeToString(effects.get(11n)!), expectedEffect)
   })
 
+  it('keeps track of substitutions with nested defs', () => {
+    const quintModule = buildModuleWithDefs([
+      'pure def a(x) = and{' +
```

Comment:
This is the smallest example I could find to reproduce a scenario where subsititutions where composed the wrong way and the final signature would contain non-quantified variables for no reason. This is based on `ics23`'s `leftBranchesAreEmpty`.